packages/replay/package.json

```diff
@@ -2,15 +2,11 @@
   "name": "@replayio/replay",
   "version": "0.21.4",
   "description": "CLI tool for uploading and managing recordings",
-  "bin": {
-    "replay": "bin/replay.js"
-  },
-  "main": "src/main.js",
+  "bin": "./dist/bin/replay.js",
+  "main": "./dist/src/main.js",
   "scripts": {
-    "if:exists": "node -e \"require('fs').existsSync(process.argv[1]) ? require('child_process').spawnSync(process.argv[2], process.argv.slice(3), {shell: true, stdio: 'inherit'}) : 0\"",
-    "if:source": "npm run if:exists -- src/main.ts",
-    "prepare": "npm run if:source -- npm run build",
-    "build": "rm -rf dist/ && tsc && chmod 755 dist/bin/* && cp package.json README.md dist/",
+    "prepare": "yarn run build",
+    "build": "rm -rf dist/ && tsc && chmod 755 dist/bin/*",
```

Comment:
> The build directory changes break the version command which is trying to resolve the package.json in packages/replay/src/utils.ts

Great catch and thanks for pointing this out. The easiest solution would be to temporarily bring back `package.json` copies in `dist`. I think I should get away here with package self-referencing though and I'm gonna try to use it in a moment. This relies on using `package.json#exports` but I think it should be fine here. As long as we stick to `require`-ing those `package.json`s then it won't even matter for TS. Alternatively, `find-up` could be used here but I think that the `package.json#exports` solution is gonna be the best

Reply:
Another solution would be to emit `lib` as a sibling to the root's `src` - instead of emitting `dist/src`. It has some implications on how `bin` is currently not in `src` and it's part of the same TS compilation. So I prefer not to do it right now but it's certainly something I'd like to simplify later too.

Reply:
self-referencing implemented in: https://github.com/replayio/replay-cli/pull/326/commits/614be04db0a5ee2a0141cc5aa1fdb0ad0a0e8153